packages/esm-patient-appointments-app/src/hooks/useAppointmentService.tsx

```diff
@@ -0,0 +1,17 @@
+import useSWR from 'swr';
+import { useMemo } from 'react';
+import { Service } from '../types';
+import { openmrsFetch } from '@openmrs/esm-framework';
+
+const useAppointmentService = () => {
+  const { data, error } = useSWR<{ data: Array<Service> }, Error>(
+    '/ws/rest/v1/appointmentService/all/full',
+    openmrsFetch,
+  );
+
+  return useMemo(() => {
+    return { data, error };
+  }, [error, data]);
```

Comment:
We're close!

There should be no need to memo these values though, you should really just be able to write:

```ts
const useAppointmentService = () => {
  return useSWR<{ data: Array<Service> }>(
    '/ws/rest/v1/appointmentService/all/full',
    openmrsFetch,
  );
}
```

react-swr will take care to eliminate any unnecessary renders.

Reply:
done